daffodil-test/src/test/scala/org/sgoodudfs/example/StringFunctions/Reverse.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sgoodudfs.example.StringFunctions
+
+import org.apache.daffodil.udf.UserDefinedFunction
+import org.apache.daffodil.udf.UserDefinedFunctionIdentification
+import com.sun.xml.internal.bind.v2.runtime.IllegalAnnotationsException
```

Comment:
Can this exception be removed? Since these are good examples, these should be perfect. And an import of someting internal seems odd.

Also, I'd recommend combining the example scala UDF's and provider in a single file. I think in the real world that's how people should organize their scala UDFs so we should show that in our example. Java's requirement of splitting out classes into separate files is annoying, especially to people familar with Scala.